test/core/test/test-extend.test.ts

```diff
@@ -384,3 +384,63 @@ describe('asynchronous setup/teardown', () => {
     ])
   })
 })
+
+describe('scoping variables to suite', () => {
+  const testAPI = test.extend<{
+    dependency: string
+    pkg: { dependency: string }
+  }>({
+    dependency: 'default',
+    pkg: ({ dependency }, use) => use({ dependency }),
+  })
+
+  testAPI('uses default values', ({ pkg }) => {
+    expect(pkg).toEqual({ dependency: 'default' })
+  })
+
+  describe('override dependency', () => {
+    testAPI.scoped({ dependency: 'new' })
```

Comment:
Type check/completion doesn't seem to be working, but is that expected?

Reply:
I'm not that familiar with `test.extend` implementation, but this will fix types:

```diff
export type TestAPI<ExtraContext = object> = ChainableTestAPI<ExtraContext> &
  ExtendedAPI<ExtraContext> & {
    extend: <T extends Record<string, any> = object>(
      fixtures: Fixtures<T, ExtraContext>
    ) => TestAPI<{
      [K in keyof T | keyof ExtraContext]: K extends keyof T
        ? T[K]
        : K extends keyof ExtraContext
          ? ExtraContext[K]
          : never;
    }>
-    scoped: <T extends Record<string, any> = object>(
-      fixtures: Fixtures<T, ExtraContext>
-    ) => void
+    scoped: (
+      fixtures: Fixtures<Partial<ExtraContext>>
+    ) => void
  }
```

![image](https://github.com/user-attachments/assets/68a827c6-1cc1-4045-9529-c05e7342f828)

Reply:
Pushed proposed change, should work better now